### DISCUSSION TITLE
How to set blocks and reserve ammo for a player?

### DISCUSSION THREAD

**Discussion by lokka30** [piqueserver/piqueserver#616] [Q&A] [answered]
Hi all!
I run The Hallway, and we are looking for a way to set the blocks and reserve ammo for a player using Python.

We are looking to program an 'infinite blocks' script, but we can't adjust the player's blocks amount without running refill(), which we do not want to do since it also refills grenades and health.

We are also looking to program an 'infinite reserve ammo' script which allows you to reload your weapon infinitely.

If it is not possible to set a player's block count or reserve ammo count, then would this be added to piqueserver in the future?

Thanks. :)

**Comment by NotAFile** [comment, marked as answer]
Unfortunately the protocol does not support this. You can only replenish the full player.

There's some scripts that refill the player and then apply damage afterwards and manually keep track of health, but they're not very pretty or intuitive.

**Reply by lokka30** [reply]
Ah, that's a shame. Thank you very much for the comment (and, for piqueserver!). Go well!
